src/App.tsx

```diff
@@ -1,8 +1,60 @@
-import React from 'react';
+import React, {
+  FC, useState, useMemo, /* useCallback, */
+} from 'react';
+/* import { debounce } from 'lodash'; */
 import './App.css';
+import { getPreparedPosts } from './api/utils';
+import { PostsList } from './components/PostList/PostList';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+export const App: FC = () => {
+  const [isLoading, setIsLoadind] = useState(false);
+  const [posts, setPosts] = useState<PostsInterface[]>([]);
+  const [query, setQuery] = useState('');
 
-export default App;
+  const handleLoadButton = async () => {
+    setIsLoadind(true);
+    const postsFromServer = await getPreparedPosts();
+
+    setPosts(postsFromServer);
+  };
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
```

Comment:
You can destructure event.target.value. Probably it's optional but i would change that.